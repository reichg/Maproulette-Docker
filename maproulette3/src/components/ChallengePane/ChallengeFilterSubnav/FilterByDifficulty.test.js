import React from 'react'
import { FilterByDifficulty } from './FilterByDifficulty'
import _cloneDeep from 'lodash/cloneDeep'

const propsFixture = {
  user: {
    id: 11,
    savedChallenges: [],
  },
  challenges: [
    {
      id: 309,
      name: "Challenge 309",
      blurb: "Challenge 309 blurb",
      description: "Challenge 309 description",
      parent: {
        displayName: "foo",
      }
    },
    {
      id: 311,
      name: "Challenge 311",
      blurb: "Challenge 311 blurb",
      description: "Challenge 311 description",
      parent: {
        displayName: "bar",
      }
    }
  ],
  searchFilters: {
    difficulty: 1
  }
}

let basicProps = null

beforeEach(() => {
  basicProps = _cloneDeep(propsFixture)
  basicProps.setSearchFilters = jest.fn()
  basicProps.removeSearchFilters = jest.fn()
  basicProps.intl = {formatMessage: jest.fn(m => m.defaultMessage)}
})

test("it renders with props as expected", () => {
  const wrapper = shallow(
    <FilterByDifficulty {...basicProps} />
  )

  expect(wrapper).toMatchSnapshot()
})

test("it calls setSearchFilters if an onChange occurs with a value", () => {
  const wrapper = shallow(
    <FilterByDifficulty {...basicProps} />
  )

  wrapper.instance().updateFilter({value: 'hard'})
  expect(basicProps.setSearchFilters).toBeCalledWith({"difficulty": "hard"})
})

test("it calls removeSearchFilters if an onChange occurs with a null value", () => {
  const wrapper = shallow(
    <FilterByDifficulty {...basicProps} />
  )

  wrapper.instance().updateFilter({value: null})
  expect(basicProps.removeSearchFilters).toBeCalledWith(['difficulty'])
})
