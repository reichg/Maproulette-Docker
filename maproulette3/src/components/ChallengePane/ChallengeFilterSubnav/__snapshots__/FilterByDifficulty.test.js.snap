// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders with props as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterByDifficulty
    challenges={
      Array [
        Object {
          "blurb": "Challenge 309 blurb",
          "description": "Challenge 309 description",
          "id": 309,
          "name": "Challenge 309",
          "parent": Object {
            "displayName": "foo",
          },
        },
        Object {
          "blurb": "Challenge 311 blurb",
          "description": "Challenge 311 description",
          "id": 311,
          "name": "Challenge 311",
          "parent": Object {
            "displayName": "bar",
          },
        },
      ]
    }
    intl={
      Object {
        "formatMessage": [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "defaultMessage": "Easy",
                "id": "Challenge.difficulty.easy",
              },
            ],
            Array [
              Object {
                "defaultMessage": "Normal",
                "id": "Challenge.difficulty.normal",
              },
            ],
            Array [
              Object {
                "defaultMessage": "Expert",
                "id": "Challenge.difficulty.expert",
              },
            ],
            Array [
              Object {
                "defaultMessage": "Any",
                "id": "Challenge.difficulty.any",
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": "Easy",
            },
            Object {
              "isThrow": false,
              "value": "Normal",
            },
            Object {
              "isThrow": false,
              "value": "Expert",
            },
            Object {
              "isThrow": false,
              "value": "Any",
            },
          ],
        },
      }
    }
    removeSearchFilters={[MockFunction]}
    searchFilters={
      Object {
        "difficulty": 1,
      }
    }
    setSearchFilters={[MockFunction]}
    user={
      Object {
        "id": 11,
        "savedChallenges": Array [],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "button": <ButtonFilter
        selection={
          <FormattedMessage
            defaultMessage="Easy"
            id="Challenge.difficulty.easy"
            values={Object {}}
          />
        }
        type={
          <FormattedMessage
            defaultMessage="Difficulty"
            id="ChallengeFilterSubnav.filter.difficulty.label"
            values={Object {}}
          />
        }
      />,
      "children": <ListDifficultyTypes
        difficultyLabels={
          Object {
            "any": "Any",
            "easy": "Easy",
            "expert": "Expert",
            "normal": "Normal",
          }
        }
        updateFilter={[Function]}
      />,
      "className": "mr-dropdown--flush xl:mr-border-l xl:mr-border-white-10 mr-p-6 mr-pl-0 xl:mr-pl-6",
      "close": [Function],
      "isVisible": false,
      "toggleVisible": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "difficultyLabels": Object {
          "any": "Any",
          "easy": "Easy",
          "expert": "Expert",
          "normal": "Normal",
        },
        "updateFilter": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "button": <ButtonFilter
          selection={
            <FormattedMessage
              defaultMessage="Easy"
              id="Challenge.difficulty.easy"
              values={Object {}}
            />
          }
          type={
            <FormattedMessage
              defaultMessage="Difficulty"
              id="ChallengeFilterSubnav.filter.difficulty.label"
              values={Object {}}
            />
          }
        />,
        "children": <ListDifficultyTypes
          difficultyLabels={
            Object {
              "any": "Any",
              "easy": "Easy",
              "expert": "Expert",
              "normal": "Normal",
            }
          }
          updateFilter={[Function]}
        />,
        "className": "mr-dropdown--flush xl:mr-border-l xl:mr-border-white-10 mr-p-6 mr-pl-0 xl:mr-pl-6",
        "close": [Function],
        "isVisible": false,
        "toggleVisible": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "difficultyLabels": Object {
            "any": "Any",
            "easy": "Easy",
            "expert": "Expert",
            "normal": "Normal",
          },
          "updateFilter": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
